File: src/main/scala/com/vc/easy/L276.scala
package com.vc.easy

object L276 {
  def numWays(n: Int, k: Int): Int = {
    if(n == 0) 0
    else if(n == 1) k
    else {
      var same = k
      var diff = k * (k - 1)
      (2 until n).foreach(x => {
        val total = diff + same
        same = diff
        diff = total * (k - 1)
      })
      same + diff
    }
  }
}

File: src/main/java/com/vc/easy/L276.java
package com.vc.easy;

class L276 {
    public int numWays(int posts, int color) {
        if(posts == 0) return 0;
        if(posts == 1) return color;

        int diff = color * (color - 1);
        int same = color;
        int total = 0;

        for(int i = 2; i <= posts; i++) {
            total = same + diff;
            same = diff;
            diff = total * (color - 1);
        }
        return total;
    }
}
